README.md

```diff
@@ -101,6 +101,8 @@ their default values indicated below:
     cmake -DCMAKE_INSTALL_PREFIX:PATH="/usr/local" \  # Installation prefix
         -DCMAKE_BUILD_TYPE:STRING="Release" \         # Type of build, one of: Debug or Release
         -DWITH_BFD:BOOL=OFF \                         # Install with BFD library (requires binutils-dev)s
+        -DWITH_COTIRE=ON \                            # Use Cotire for precompiled headers
+        -DWITH_CPP14=OFF \                            # Enable C++14 during compilation
```

Comment:
I was hoping to remove this option before. This is needed for LLVM 10 right?

Reply:
Yeah, that's my understanding. Requiring C++14 sounds reasonable to me.

Reply:
Requiring C++14 for that case is fine. I was hoping to do it automatically instead of a non-standard option like this.

Reply:
Sounds good. I'll close this PR for now then?

Reply:
WITH_COTIRE needs to be documented I guess. What was the issue with cotire?

Reply:
Here's one case (where not setting WITH_COTIRE=no causes compilation failure):
```
+-------------------------------+
| Configuration results SUMMARY |
+-------------------------------+

CMAKE_C_COMPILER:   /usr/bin/clang-10
CMAKE_CXX_COMPILER: /usr/bin/clang++-10
CMAKE_BUILD_TYPE: Debug
CMAKE_CONFIGURATION_TYPES: 
CMAKE_C_FLAGS:           
CMAKE_CXX_FLAGS:         -fsanitize=memory -stdlib=libc++ -I/opt/libcxx_msan/include -I/opt/libcxx_msan/include/c++/v1 -fno-omit-frame-pointer -fno-optimize-sibling-calls -O1 -glldb -DHAVE_GCC_ABI_DEMANGLE=no
CMAKE_CXX_FLAGS_DEBUG:   -Wall -Wextra -Wno-unused-parameter -g -ggdb -std=c++14 -fPIC -Wno-unknown-pragmas
CMAKE_CXX_FLAGS_RELEASE: -Wall -Wextra -Wno-unused-parameter  -funroll-loops -std=c++14 -fPIC -Wno-unknown-pragmas
CMAKE_INSTALL_PREFIX: /usr/local
BUILD_SHARED_LIBS: 
INTEGER_CLASS : BOOSTMP
HAVE_SYMENGINE_GMP: no
WITH_SYMENGINE_ASSERT: yes
WITH_SYMENGINE_RCP: ON
WITH_SYMENGINE_TEUCHOS: no
WITH_COTIRE: ON
WITH_GENERATE_PARSER: no
HAVE_GCC_ABI_DEMANGLE: yes
HAVE_C_FUNCTION_NOT_FUNC: no
HAVE_DEFAULT_CONSTRUCTORS: yes
HAVE_SYMENGINE_NOEXCEPT: yes
HAVE_SYMENGINE_IS_CONSTRUCTIBLE: yes
HAVE_SYMENGINE_RESERVE: yes
HAVE_SYMENGINE_STD_TO_STRING: TRUE
WITH_SYMENGINE_THREAD_SAFE: no
BUILD_TESTS: ON
BUILD_BENCHMARKS: ON
BUILD_BENCHMARKS_NONIUS: no
WITH_GMP: no
WITH_BFD: OFF
WITH_ECM: no
WITH_PRIMESIEVE: no
WITH_FLINT: no
WITH_ARB: no
WITH_MPFR: no
WITH_PIRANHA: no
WITH_LLVM: OFF
WITH_BOOST: yes
BOOST_INCLUDE_DIRS: /opt/boost_1_72_0
BOOST_LIBRARIES: 
WITH_PTHREAD: no
WITH_MPC: no
WITH_TCMALLOC: no
WITH_OPENMP: no
WITH_VIRTUAL_TYPEID: no

--------------------------------------------------------------------------------

-- Configuring done
-- Generating done
-- Build files have been written to: /mount/symengine-build
+ make -j 8
[  1%] Generating CXX unity source symengine/cotire/symengine_CXX_unity.cxx
[  1%] Generating CXX prefix source symengine/cotire/symengine_CXX_prefix.cxx
-- Result 1 scanning includes of /mount/symengine-build/symengine/cotire/symengine_CXX_unity.cxx.
[  1%] Generating CXX prefix header symengine/cotire/symengine_CXX_prefix.hxx
[  2%] Building CXX precompiled header symengine/cotire/symengine_CXX_prefix.hxx.pch
In file included from /mount/symengine-build/symengine/cotire/symengine_CXX_prefix.hxx:4:
In file included from /mount/symengine-build/symengine/cotire/symengine_CXX_prefix.cxx:5:
/opt/boost_1_72_0/boost/multiprecision/cpp_int.hpp:9:10: fatal error: 'iostream' file not found
#include <iostream>
         ^~~~~~~~~~
1 error generated.
CMake Error at /mount/symengine-403c076f1ddba0db02820f832a668b8e6a436d97/cmake/cotire.cmake:1826 (message):
  cotire: error 1 precompiling
  /mount/symengine-build/symengine/cotire/symengine_CXX_prefix.hxx.
Call Stack (most recent call first):
  /mount/symengine-403c076f1ddba0db02820f832a668b8e6a436d97/cmake/cotire.cmake:3421 (cotire_precompile_prefix_header)

```

Reply:
I wonder if `CMAKE_CXX_FLAGS` is used in addition to `CMAKE_CXX_FLAGS_DEBUG` both with and without Cotire?

Reply:
(the output is from [this gist](https://gist.github.com/bjodah/82e1013e2025003e35f365b89921ddc2) but with `-DWITH_COTIRE=ON`)

Reply:
Can you run `make VERBOSE=1`?

Reply:
Looks like it selectively stripped the `-I` for libcxx, but kept e.g. `-fsanitizer` som CXXFLAGS....
```
+ make VERBOSE=1 -j 8
/usr/bin/cmake -S/mount/symengine-403c076f1ddba0db02820f832a668b8e6a436d97 -B/mount/symengine-build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /mount/symengine-build/CMakeFiles /mount/symengine-build/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/mount/symengine-build'
make -f symengine/CMakeFiles/symengine.dir/build.make symengine/CMakeFiles/symengine.dir/depend
make[2]: Entering directory '/mount/symengine-build'
[  1%] Building CXX precompiled header symengine/cotire/symengine_CXX_prefix.hxx.pch
cd /mount/symengine-403c076f1ddba0db02820f832a668b8e6a436d97/symengine && /usr/bin/cmake -DCOTIRE_BUILD_TYPE:STRING=Debug -DCOTIRE_VERBOSE:BOOL=1 -P /mount/symengine-403c076f1ddba0db02820f832a668b8e6a436d97/cmake/cotire.cmake precompile /mount/symengine-build/symengine/symengine_CXX_cotire.cmake /mount/symengine-build/symengine/cotire/symengine_CXX_prefix.hxx /mount/symengine-build/symengine/cotire/symengine_CXX_prefix.hxx.pch add.cpp
-- execute_process: /usr/bin/clang++-10;-DHAVE_GCC_ABI_DEMANGLE=no;-fsanitize=memory;-stdlib=libc++;-fno-omit-frame-pointer;-fno-optimize-sibling-calls;-O1;-glldb;-Wall;-Wextra;-Wno-unused-parameter;-g;-ggdb;-std=c++14;-fPIC;-Wno-unknown-pragmas;-I/mount/symengine-403c076f1ddba0db02820f832a668b8e6a436d97;-I/mount/symengine-build;-I/opt/boost_1_72_0;-x;c++-header;-c;/mount/symengine-build/symengine/cotire/symengine_CXX_prefix.hxx;-o;/mount/symengine-build/symengine/cotire/symengine_CXX_prefix.hxx.pch
In file included from /mount/symengine-build/symengine/cotire/symengine_CXX_prefix.hxx:4:
In file included from /mount/symengine-build/symengine/cotire/symengine_CXX_prefix.cxx:5:
/opt/boost_1_72_0/boost/multiprecision/cpp_int.hpp:9:10: fatal error: 'iostream' file not found
#include <iostream>
         ^~~~~~~~~~
1 error generated.
CMake Error at /mount/symengine-403c076f1ddba0db02820f832a668b8e6a436d97/cmake/cotire.cmake:1826 (message):
  cotire: error 1 precompiling
  /mount/symengine-build/symengine/cotire/symengine_CXX_prefix.hxx.
Call Stack (most recent call first):
  /mount/symengine-403c076f1ddba0db02820f832a668b8e6a436d97/cmake/cotire.cmake:3421 (cotire_precompile_prefix_header)

```

Reply:
I thought `cotire` was not supposed to do anything like that. In that case, I'd be fine with changing the default to off for cotire.

Reply:
Latest CMake supports unity builds I believe, so I guess cotire is pretty much deprecated anyhow?

Reply:
sorry, I meant latest cmake supports precompiled headers...

Reply:
(using latest cmake is always trivial since they provide such nice stand-alone binaries which even runs on ancient systems)

Reply:
(except CentOS 5 which I used for manylinux1 builds. Should move to manylinux2010)

Reply:
Oh, didn't know those were CentOS 5 based.